docs/how-chugsplash-works.md

```diff
@@ -0,0 +1,252 @@
+2. Approval: The proposed deployment or upgrade is approved by the project's owner, which is usually governance or a multisig. Approvals occur as a single tiny transaction regardless of the size of the deployment.
+3. Execution: As soon as a deployment or upgrade is approved, it's completed in this step. This usually takes just a couple minutes regardless of the size of the deployment or upgrade.
+
+These three steps can be abstracted into a single `deploy` command for users who don't need a proposal or approval step.
+
+All of this logic is contained in your project's `ChugSplashManager` contract, which is basically a `ProxyAdmin` contract on steroids. The `ChugSplashManager` owns your proxies and is responsible for upgrading them. In turn, you're the owner of the `ChugSplashManager`.
+
+In the next few sections, we'll describe exactly how ChugSplash works by walking through a simple deployment. It's worth noting that the same exact process occurs for both deployments and upgrades.
+
+### Defining a Deployment
+
+Say you'd like to deploy the following contract using ChugSplash:
+
+```sol
+contract MyContract {
+  uint256 public myVariable;
+}
+```
+
+You begin by creating a ChugSplash file, which contains all of the information necessary to deploy or upgrade a project. The ChugSplash file for this contract would look something like:
+
+```ts
+{
+  MyContract: {
+    myVariable: 1234
+  },
+  ... // config options
+}
+```
+
+### Proposal
+
+During the proposal step, the ChugSplash file is first converted into a format that can be executed on-chain. There are two components: the contract's creation bytecode and its state variables.
+
+The state variable definitions in the ChugSplash file are encoded into a series of 32-byte storage slot key/value pairs:
+
+```
+[0x000...000, 0x000...04D2]
+```
+
+This encoding occurs off-chain using the contract's storage layout.
+
+> At this point, you might be thinking, "*why would they compile the variables into individual storage slot values instead of just calling existing functions on the contract?*"\
+The short answer is that it allows us to guarantee your deployment is deterministic, meaning it can't halt for any reason. Head to the [FAQ section](#why-does-chugsplash-encode-variables-into-storage-slot-values-instead-of-just-calling-existing-functions-on-the-contract) for further explanation.
+
+It's feasible to approve a small deployment in a single transaction from a multisig or governance, but this quickly becomes unrealistic for larger deployments. One solution would be to use multicall from a multisig, which batches smaller transactions into a single larger one. However, multicall transactions are capped at the block gas limit, so this doesn't solve the original problem of *click* *click* *clicking* on a Ledger for 30 minutes during a large deployment. Additionally, it becomes difficult to verify the correctness of these enormous transactions on a tiny Ledger screen or a multisig UI.
```

Comment:
It's pretty confusing to jump to talking about approving with a multisig in the section on proposals. I get why you're talking about it though b/c it's important to explain the reasoning behind the architecture in order to explain how the proposal step works. However, I wouldn't include that explanation in this section. I would move it up to the beginning where we talk about the three steps. I think that section could be a lot longer and include all the details about why those three steps exist, what the properties of them are, and how they relate to each other.